samples/bluetooth/direct_test_mode/src/fem/nrf21540.c

```diff
@@ -0,0 +1,526 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <device.h>
+#include <drivers/gpio.h>
+#include <drivers/spi.h>
+#include <soc.h>
+#include <sys/__assert.h>
+
+#include <nrfx/hal/nrf_gpio.h>
+#include <nrfx/hal/nrf_gpiote.h>
+#include <nrfx/hal/nrf_radio.h>
+#include <nrfx/helpers/nrfx_gppi.h>
+#include <nrfx_timer.h>
+
+#if defined(CONFIG_HAS_HW_NRF_PPI)
+#include <nrfx_ppi.h>
+#define gppi_channel_t nrf_ppi_channel_t
+#define gppi_channel_alloc nrfx_ppi_channel_alloc
+#elif defined(CONFIG_HAS_HW_NRF_DPPIC)
+#include <nrfx_dppi.h>
+#define gppi_channel_t uint8_t
+#define gppi_channel_alloc nrfx_dppi_channel_alloc
+#else
+#error "No PPI or DPPI"
+#endif
```

Comment:
Suggest moving this logic to the nrfx_gppi. It fits quite well in GPPI concept which is an abstraction over PPI/DPPI.

Reply:
It can be done in a separated PR and next aligned here.